### PR TITLE
Fix to check if session is valid after creation

### LINKED ISSUE #712: findCollections and createCollection do not show the correct error message when the Astra token is invalid
When an Astra token is used in the wrong database context, there is an issue with the connection handling. For example, if `Token A`, which is created for use with Database `A`, is mistakenly used for Database `B`, the connection should be refused. However, currently, the connection is established successfully, and it results in an error message `Unknown namespace`. This is misleading as the real issue is the use of an invalid token for the given database. Therefore, the error message should be changed to `Invalid token`.

The current error message is:
```
{
    "errors": [
        {
            "message": "INVALID_ARGUMENT: Unknown namespace 'default_keyspace', you must create it first.",
            "errorCode": "NAMESPACE_DOES_NOT_EXIST"
        }
    ]
}
```
The error message should be:
```
{
    "errors": [
        {
            "message": "UNAUTHENTICATED: Invalid token"
        }
    ]
}
```

### PR DESCRIPTION
**Pull request by kathirsvn**
**What this PR does**:
Adding fix to validate Astra CQL session after getting created.

**Which issue(s) this PR fixes**:
Fixes #712 

**Checklist**
- [X] Changes manually tested
- [X] Automated Tests added/updated
- [X] Documentation added/updated
- [X] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Review by mpenick — APPROVED** [review]

File: src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CQLSessionCache.java [hunk 1]
```diff
@@ -151,18 +151,54 @@ private CqlSession getNewSession(SessionCacheKey cacheKey) {
...
+                  TOKEN,
+                  Objects.requireNonNull(((TokenCredentials) cacheKey.credentials()).token()))
+              .withLocalDatacenter(operationsConfig.databaseConfig().localDatacenter())
+              .withClassLoader(Thread.currentThread().getContextClassLoader())
+              .withApplicationName(APPLICATION_NAME)
+              .withConfigLoader(loader)
+              .build();
+      if (!isAstraSessionValid(cqlSession, cacheKey.tenantId())) {
+        throw new UnauthorizedException(
+            "Unauthorized to access tenant %s's data".formatted(cacheKey.tenantId()));
```
Comment (JeremiahDJordan) [inline src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CQLSessionCache.java:166]: don't use the word "tenant" in anything user facing.  Also don't put the ID in the message.  There error message here should be no different from the error thrown when a bad token is used.

File: src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CQLSessionCache.java [hunk 2]
```diff
@@ -151,18 +151,54 @@ private CqlSession getNewSession(SessionCacheKey cacheKey) {
...
+   */
+  private boolean isAstraSessionValid(CqlSession cqlSession, String tenantId) {
+    if (!cqlSession.getMetadata().getKeyspaces().isEmpty()) {
+      return true;
+    }
+    try {
+      cqlSession.execute("SELECT * FROM system_virtual_schema.tables");
+      return true;
+    } catch (com.datastax.oss.driver.api.core.servererrors.UnauthorizedException e) {
+      LOGGER.error("Unauthorized to access tenant %s's data".formatted(tenantId), e);
```
[the remark this answers is not in the document]
Comment (kathirsvn) [reply src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CQLSessionCache.java:196]: Got it. But, I believe it's not a common occurrence, so we would change it if it becomes a noise.

**Review by jeffreyscarpenter — APPROVED** [review]
Looks good

**Review by vkarpov15 — COMMENTED** [review]
LGTM

**Pull request merged**